File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character,Character> hashmap = new HashMap<>();
        for(int i=0;i<s.length();i++){
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);
            
            if(hashmap.containsKey(sChar)){
                if(tChar != hashmap.get(sChar)){
                    return false;
                }
            }else{
                if(hashmap.containsValue(tChar)){
                    return false;
                }else{
                    hashmap.put(sChar,tChar);
                }
            }
        }
        return true;
        
    }
} 

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hashmap = {}
        for i in range(len(s)):
            if s[i] in hashmap:
                if t[i] != hashmap[s[i]]:    
                    return False
            else:
                if t[i] not in hashmap.values():
                    hashmap[s[i]] = t[i]
                else:
                    return False
        return True